builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -553,11 +553,13 @@ func resourceArmVirtualMachineCreate(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
-	osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
-	if err != nil {
-		return err
+	if _, ok := d.GetOk("os_profile"); ok {
+		osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
+		if err != nil {
+			return err
+		}
+		properties.OsProfile = osProfile
 	}
```

Comment:
@tombuildsstuff What is the etiquette for pushing an update to my PR?  Do I push additional commits or should I squash it somehow?  Guidance appreciated.